indico/modules/events/layout/controllers/layout.py

```diff
@@ -65,12 +69,50 @@ def _process_lecture_meeting(self):
             return redirect(url_for('.index', self.event_new))
         return WPLayoutEdit.render_template('layout_meeting_lecture.html', self._conf, form=form, event=self.event_new)
 
+    def _make_weeks_view_form(self):
+        form_class = type(b'WeeksViewForm', (IndicoForm,), {})
+        fields = theme_settings.themes['indico-weeks-view']['arguments']
+        defaults = {}
+        for field_name in fields:
+            field_type = fields[field_name]['type']
+            field_class = getattr(indico_fields, field_type, None) or getattr(wtforms_fields, field_type, None)
+            if not field_class:
+                continue
+            field_object = type(str(field_name), (field_class,), {})
+            label = fields[field_name]['caption']
+            description = fields[field_name]['description']
+            defaults[field_name] = fields[field_name]['default'] if fields[field_name].get('default') else None
+            validators = [DataRequired()] if fields[field_name].get('required') else []
+            if field_class == MultipleItemsField:
```

Comment:
Why do we need custom logic here? Ideally this is all defined from the YAML data...